src/m365/spo/commands/listitem/listitem-roleassignment-add.spec.ts

```diff
@@ -0,0 +1,421 @@
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation neither listTitle nor listId or listUrl is specified', async () => {
+    const actual = await command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', groupName: 'someGroup', listItemId: 1, roleDefinitionName: 'readers' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if the listItemId option is not a number', async () => {
+    const actual = await command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', listTitle: 'Documents', groupName: 'someGroup', listItemId: 'abc', roleDefinitionName: 'readers' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if the listItemId option is a number', async () => {
+    const actual = await command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', listTitle: 'Documents', groupName: 'someGroup', listItemId: 1, roleDefinitionName: 'readers' } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('add role assignment to listitem in list by title and role definition id', async () => {
+    sinon.stub(request, 'post').callsFake((opts) => {
+      if ((opts.url as string).indexOf('_api/web/lists/getByTitle(\'test\')/items(1)/roleassignments/addroleassignment(principalid=\'11\',roledefid=\'1073741827\')') > -1) {
+        return Promise.resolve();
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    await command.action(logger, {
+      options: {
+        debug: true,
+        webUrl: 'https://contoso.sharepoint.com',
+        listTitle: 'test',
+        listItemId: 1,
+        principalId: 11,
+        roleDefinitionId: 1073741827
+      }
+    });
+  });
+
+  it('add role assignment to listitem in list by id and role definition id', async () => {
+    sinon.stub(request, 'post').callsFake((opts) => {
+      if ((opts.url as string).indexOf('/_api/web/lists(guid\'0CD891EF-AFCE-4E55-B836-FCE03286CCCF\')/items(1)/roleassignments/addroleassignment(principalid=\'11\',roledefid=\'1073741827\')') > -1) {
+        return Promise.resolve();
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    command.action(logger, {
+      options: {
+        debug: true,
+        webUrl: 'https://contoso.sharepoint.com',
+        listItemId: 1,
+        listId: '0CD891EF-AFCE-4E55-B836-FCE03286CCCF',
+        principalId: 11,
+        roleDefinitionId: 1073741827
+      }
+    });
+  });
+
+  it('add role assignment to listitem in list by url and role definition id', async () => {
+    sinon.stub(request, 'post').callsFake((opts) => {
+      if ((opts.url as string).indexOf('/_api/web/GetList(\'%2Fsites%2Fdocuments\')/items(1)/roleassignments/addroleassignment(principalid=\'11\',roledefid=\'1073741827\')') > -1) {
```

Comment:
Let's try to avoid the usage of `indexOf` and in place use a full comparison.
`if (opts.url === 'exampleurl')`